src/firestore.rules.test.js

```diff
@@ -26,6 +26,21 @@ const authedApp = initializeTestApp({
 const unauthedApp = initializeTestApp({
   projectId: MY_PROJECT_ID,
 });
+const authedApp2 = initializeTestApp({
+  projectId: MY_PROJECT_ID,
+  auth: { uid: 'john-doe' },
+});
+
+const converter = {
+  toFirestore: (data) => {
+    const user = { uid: 'alice', email: 'test' };
+    const author =
+      data.author ?? (user ? { id: user?.uid, email: user?.email } : {});
+
+    console.log({ ...data, author });
+    return { ...data, author };
+  },
+};
```

Comment:
just testing some things, but with this at least the user is added lol

Reply:
I suggest debugging what `firebase.app()` is as that's what gets used in the normal converter:
https://github.com/beyondhb1079/s4us/blob/eb8ba2ffe3a1b6de8f32c9135a39c1164c024a3b/src/models/Scholarships.ts#L12

Maybe add `console.log(JSON.stringify())` statements or set a debug breakpoint? (I've yet to figure out how to run the debugger for tests but I'm sure there's a way).

Reply:
Ah wait, of course it's like this! None of the tests below are using the converter defined in `Scholarship.ts`:
https://github.com/beyondhb1079/s4us/blob/eb8ba2ffe3a1b6de8f32c9135a39c1164c024a3b/src/models/Scholarships.ts#L9

They're all just simply doing doing `firestore().collection('scholarships')` instead of what `FirestoreCollection.collection` does:
https://github.com/beyondhb1079/s4us/blob/eb8ba2ffe3a1b6de8f32c9135a39c1164c024a3b/src/models/base/FirestoreCollection.ts#L10-L15

Make sure to pass `.withConverter(Scholarships.converter)` everywhere you need to rely on converter logic.

Reply:
Actually, you know what, it might still not work because there are _multiple_ apps and `Scholarships.ts` always uses the _default_ one (which I'm not sure how it's determined). I think we need to separate the Firebase app from these collection classes so that we can have multiple handles using different apps. Ideally you'd be able to do something like:
```js
const app1Scholarships = Scholarships.connect(app1);
const app2Scholarships = Scholarships.connect(app2);

app1Scholarships.new()...
app2Scholarships.new()...
```

Feel free to skip getting the second test to work (you can comment it out with a TODO) and file an issue to support specifying the Firestore connection for our Firestore collections.